### PR TITLE
feat: use `got` instead of `request`

### PR DESCRIPTION
**Pull request by BarryCarlyon**
As per

https://github.com/request/request/issues/3142

Request is deprecated/won't be updated.

This PR replaces Request with Got as simply as possible. (and will save a dependancy for people whom moved from Request to other libs)

All the tests pass (after local hacking vows to work as it calls dead util's functions)

### CODE REVIEW

**Comment by brunobg** [comment]
Reviewed the code and patch seems fine. This comment should run the tests again and if passes I suggest it to be accepted.

**Comment by brunobg** [comment]
ahn, the comment didn't trigger travis, I guess because I'm not a maintainer? @Kikobeats can you trigger this?

**Comment by Kikobeats** [comment]
@BarryCarlyon @brunobg I trigger a new build but tests are still failing

Can you resolve the conflict?

**Comment by brunobg** [comment]
Yes, I'll check this later.

**Comment by BarryCarlyon** [comment]
So I'm poking about to fix.

I ran the tests, and ones similar to travis failed.

I added a console.log somewhere and ran the tests again and got a

> ✗ Broken » 20 honored ∙ 1 broken (29.126s)

Broken test on `thumbnail was downloaded`

Some inconsistencies somewhere?! (node 14 on my local box)

**Comment by BarryCarlyon** [comment]
```
~/Sites/github/node-youtube-dl(replacerequest) $ npm test

> youtube-dl@3.0.2 test /Users/barrycarlyon/Sites/github/node-youtube-dl
> vows ./test/*.js --spec


  ♢ download

  a video with no format specified
    ✓ data returned
    ✓ file was downloaded
  a video with video format specified
    ✓ data returned
    ✓ file was downloaded
  a video with thumbnail
    ✓ thumbnail was downloaded
  a video with subtitles
    ✓ subtitles were downloaded
  a video with audio format specified
    ✓ data returned
    ✓ file was downloaded
  a video with filesize: null
    ✓ data returned
    ✓ file was downloaded

  ♢ getExtractors

  plain extractors
    ✓ extractors returned
  extractors with description
    ✓ extractors returned

  ♢ getInfo

  from a twitch video
    ✓ info returned
  with get url only flag
    ✓ info returned
  from a youtube video
    ✓ info returned
  from multiple videos
    ✓ info returned
  from a youtube playlist
    ✓ info returned

  ♢ download playlist

  thumbnails of a playlist
    ✓ data returned
    ✓ thumbnails were downloaded
  from a youtube playlist
    ✓ data returned
    ✓ files downloaded

✓ OK » 21 honored (31.556s)
```

All good here, we'll see what travis does!

**Comment by Kikobeats** [comment]
@BarryCarlyon can you resolve the conflict? 🙂

**Comment by BarryCarlyon** [comment]
A block of code I wrote replaces the block of code that is conflicting.

I think this is right? First time I've had to do this github side

File: package.json [hunk 1]
```diff
@@ -181,9 +181,9 @@
   "dependencies": {
     "debug": "~4.1.1",
     "execa": "~3.2.0",
+    "got": "~11.6.0",
     "hh-mm-ss": "~1.2.0",
     "mkdirp": "~0.5.1",
-    "request": "~2.88.0",
     "streamify": "~0.2.9",
```
Comment (Kikobeats) [inline package.json:187]: @BarryCarlyon maybe this dependency is no more necessary?

got has built-in stream support

Comment (BarryCarlyon) [reply package.json:187]: I missed that when I was having a mare with git commit-ing

Comment (BarryCarlyon) [reply package.json:187]: You on about line 187?

I thought you were saying I had lleft a request dep in?

Comment (BarryCarlyon) [reply package.json:187]: I'll take a look see and see what else needs to be changed when I get a second

**Comment by Kikobeats** [comment]
I updated the PR but I'm not sure why Travis is still failing.

All the tests are green on my local using different Node.js versions.

so I guess maybe the URL that tests are targeting is denying the access when the IP address comes from Travis servers?

In any case, I want to move this forward so I think could be enough safe ship as a new minor version

**Comment by BarryCarlyon** [comment]
> so I guess maybe the URL that tests are targeting is denying the access when the IP address comes from Travis servers?

Classic YouTube/Google. They seem to block known datacenter IP ranges quite often

But glad we finally got there on this. (Joys of being too busy RL to come back and help)

**Pull request merged**